### PR TITLE
Add exclude to options

### LINKED ISSUE #12: New Feature For Skipping Specified Char
Recently I was working on one project and I was asked to generate a random String with numeric and letter values with specified number like 0 and specified char O as my client was getting confused in between this two char. 

So I am suggesting here to add one more argument to the function with array specified value like as below.
[ '0', 'O', 'o'] so it will generate a random string without given values in array.

### PR DESCRIPTION
**Pull request by kivlor**
Fixes #12

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -79,6 +79,12 @@ function randomatic(pattern, length, options) {
   if (pattern.indexOf('*') !== -1) mask += type.all;
   if (custom) mask += pattern;
 
+  // Characters to exclude
+  if (opts.exclude) {
+    var exclude = typeOf(length) === 'string' ? opts.exclude : opts.exclude.join('');
+    mask = mask.replace(new RegExp('['+exclude+']+', 'g'), '');
```
Comment (doowb) [inline index.js]: Please add spaces around the plus signs in this line... `'[' + exclude + ']+'`

**Comment by doowb** [comment]
@kivlor thanks! Looks good. I just had one syntax comment, then it looks good to me.

From #12:

> I wasn't sure the best approach to take to allow passing a function

I think this means to check if `exclude` is a function just before checking for `string` or `array` and calling `exclude` passing in the current `mask`.

At first, I thought it meant taking a function that allowed returning the `mask`, but that can be done by passing a string on `options.chars`.

I think with this change, a scenario that I thought of will be addressed: I want a random string that contains most of the characters provided here (all - some excluded chars) + some chars that aren't included.

```js
const id = randomatic('?*', 20, { chars: '|', exclude: ['1lL0oO.'] });
```

**Comment by kivlor** [comment]
@doowb syntax comment addressed!

Thanks for explaining my 'passing a function' question. Seems with out it this change will still work (and without the complexity).

**Comment by doowb** [comment]
Thanks @kivlor this has been merged and published as `3.1.0`.

**Pull request merged**